### DISCUSSION TITLE
Issue with plugin install after 'npx graphql-codegen init'

### DISCUSSION THREAD

**Discussion by danbiagini** [dotansimha/graphql-code-generator#4644] [General] [answered]
Hello,

Newbie, but this seems like its a bug.  I installed codegen, ran init and then ran 'npm install' as instructed however the plugin packages are failing to install.  Looks like the versions added to package.json are wrong... 

`    "@graphql-codegen/typescript": "~1.17.8",
    "@graphql-codegen/typescript-operations": "~1.17.8",
    "@graphql-codegen/typescript-react-apollo": "~1.17.8",
    "@graphql-codegen/introspection": "~1.17.8"
`
Seems there's no version typescript-react-apollo: 1.17.8

**Repro Steps:**

1. `npm install -D @graphql-codegen/cli`
2. `npx graphql-codegen init

    Welcome to GraphQL Code Generator!
    Answer few questions and we will setup everything for you.

? What type of application are you building? Application built with React
? Where is your schema?: (path or url) schema.graphql
? Where are your operations and fragments?: src/**/*.graphql
? Pick plugins: TypeScript (required by other typescript plugins), TypeScript Operations (operations and fragments), TypeScript React Apollo (typed components and HOCs)
? Where to write the output: src/generated/graphql.tsx
? Do you want to generate an introspection file? Yes
? How to name the config file? codegen.yml
? What script in package.json should run the codegen? gql-gen

    Config file generated at codegen.yml

      $ npm install

    To install the plugins.

      $ npm run gql-gen

    To run GraphQL Code Generator.
`
3. `cat codegen.yml
overwrite: true
schema: "schema.graphql"
documents: "src/**/*.graphql"
generates:
  src/generated/graphql.tsx:
    plugins:
      - "typescript"
      - "typescript-operations"
      - "typescript-react-apollo"
  ./graphql.schema.json:
    plugins:
      - "introspection"`
4. `npm install
npm WARN deprecated core-js@2.6.11: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm ERR! code ETARGET
npm ERR! notarget No matching version found for @graphql-codegen/typescript-react-apollo@1.17.8.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.`

**Comment by sin-to-jin** [comment]
Hi, other packages that depend on `@graphql-codegen/cli` latest don't seem to be published in the https://www.npmjs.com/ version yet.

https://www.npmjs.com/package/@graphql-codegen/typescript-react-apollo/v/1.17.8
Like this one, since it's a 404.

The quickest way is to drop one patch version of `package.json`.
If you change `1.17.8` to `1.17.7` and `npm intall` it will install successfully. In that case, it seems to be good to upgrade when it is published.

**Comment by dotansimha** [comment]
Hi!
We recently moved to a more flexible way for versioning our packages. That means, packages can now be released independently. Before that, all packages were published together, under the same version. 
The `init` command is still not well adjusted to use that form. @kamilkisiela is currently working on a fix for that. 

You can use the `init` command ,but please after that, make sure to update the plugins versions (to latest) in `package.json` until we'll fix that in the cli.

Sorry for the inconvenience !

**Comment by DmitryOlkhovoi** [comment]
@dotansimha @ogasawaraShinnosuke 

> core-js@2.6.11: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm ERR! code ETARGET
npm ERR! notarget No matching version found for @graphql-codegen/typescript-react-apollo@^1.17.8.
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
npm ERR! notarget 
npm ERR! notarget It was specified as a dependency of 'devs-content'
npm ERR! notarget

**Reply by DmitryOlkhovoi** [reply]
I see that's the same error. Downgraded to 1.17.7 and it works

**Comment by dotansimha** [comment, marked as answer]
Fixed in: https://github.com/dotansimha/graphql-code-generator/pull/4809 , published as `@graphql-codegen/cli@1.17.9`
